docs/pages/docs/guides/providers/mistral.mdx

```diff
@@ -0,0 +1,228 @@
+---
+title: Mistral
+---
+
+import { Steps, Callout } from 'nextra-theme-docs';
+
+# Mistral
+
+Vercel AI SDK provides a set of utilities to make it easy to use [Mistral](https://mistral.ai/)'s APIs and models. In this guide, we'll walk through how to use the utilities to create a chat bot and a text completion app.
+
+## Guide: Mistral Chatbot
+
+<Steps>
+
+### Create a Next.js app
+
+Create a Next.js application and install `ai` and `@mistralai/mistralai`, the Vercel AI SDK and Mistral API client respectively.
+
+````sh
+
+```sh
+pnpm dlx create-next-app my-ai-app
+cd my-ai-app
+pnpm install ai @mistralai/mistralai
+````
+
+### Add your Mistral API Key to `.env`
+
+Create a `.env` file in your project root and add your Mistral API Key:
+
+```env filename=".env"
+MISTRAL_API_KEY=xxxxxxxxx
+```
+
+### Create a Route Handler
+
+Create a Next.js Route Handler that we'll use to generate a chat completion via Mistral that we'll then stream back to our Next.js.
+
+For this example, we'll create a route handler at `app/api/chat/route.ts` that accepts a `POST` request with a `messages` array of strings:
+
+```tsx filename="app/api/chat/route.ts" showLineNumbers
+import { OpenAIStream, StreamingTextResponse } from 'ai';
+
+// Note: There are no types for the Mistral API client yet.
+// @ts-ignore
+import MistralClient from '@mistralai/mistralai';
+
+export async function POST(req: Request) {
+  // Extract the `messages` from the body of the request
+  const { messages } = await req.json();
+
+  const client = new MistralClient(process.env.MISTRAL_API_KEY || '');
+
+  const response = await client.chatStream({
+    model: 'mistral-tiny',
+    stream: true,
+    max_tokens: 1000,
+    messages,
+  });
+
+  // Convert the response into a friendly text-stream. The Mistral client responses are
+  // compatible with the Vercel AI SDK OpenAIStream adapter.
+  const stream = OpenAIStream(response);
```

Comment:
Is this correct? It is compatible with an OpenAIStream?

Reply:
yes double checked and also ran the code